engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/text/AnyPrettyNode.java

```diff
@@ -0,0 +1,86 @@
+package org.enso.interpreter.node.expression.builtin.text;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.callable.atom.AtomConstructor;
+import org.enso.interpreter.runtime.data.text.Text;
+
+@BuiltinMethod(type = "Any", name = "pretty", description = "Generic pretty printing function.")
+public abstract class AnyPrettyNode extends Node {
+  private static final int DISPATCH_CACHE = 3;
+  private @Child InteropLibrary displays =
+      InteropLibrary.getFactory().createDispatched(DISPATCH_CACHE);
+  private @Child InteropLibrary strings =
+      InteropLibrary.getFactory().createDispatched(DISPATCH_CACHE);
+
+  public static AnyPrettyNode build() {
+    return AnyPrettyNodeGen.create();
+  }
+
+  public abstract Text execute(Object self);
+
+  @Specialization
+  Text doAtom(Atom at) {
+    if (at.getFields().length == 0) {
+      return consName(at.getConstructor());
```

Comment:
This case could be on fast-path, I believe. It doesn't concatenate strings, but instantiates `Text.create`. One would have to speculate on the constructor:
```
@Cached("at.getConstructor()") AtomConstructor cachedConstructor
```
and `guards=at.getConstructor() == cachedConstructor` and `cachedConstructor.getFieldsLength() == 0`. Then one can even `@Cache("consName(cachedConstructor.getConstructor()") String name`.